analysis/mage/src/StandardChecks.tsx

```diff
@@ -0,0 +1,161 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import CASTS_THAT_ARENT_CASTS from 'parser/core/CASTS_THAT_ARENT_CASTS';
+import { SpellInfo } from 'parser/core/EventFilter';
+import {
+  EventType,
+  AnyEvent,
+  CastEvent,
+  BeginChannelEvent,
+  BeginCastEvent,
+} from 'parser/core/Events';
+import Events from 'parser/core/Events';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import EventHistory from 'parser/shared/modules/EventHistory';
+
+class StandardChecks extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    eventHistory: EventHistory,
+  };
+  protected abilityTracker!: AbilityTracker;
+  protected eventHistory!: EventHistory;
+
+  castBreakdownDuringBuff(buff: SpellInfo) {
+    /**
+     * @param buff the spell object for the buff
+     * @returns an array containing each unique spell cast and the number of times it was cast
+     */
+    const castEvents: CastEvent[] = this.getEventsDuringBuff(buff, EventType.Cast);
+    const castArray: number[][] = [];
+    castEvents &&
+      castEvents.forEach((c) => {
+        const index = castArray.findIndex((arr) => arr.includes(c.ability.guid));
+        if (index !== -1) {
+          castArray[index][1] += 1;
+        } else {
+          castArray.push([c.ability.guid, 1]);
+        }
+      });
+    return castArray;
+  }
+
+  countTotalCasts(spell: SpellInfo) {
+    return this.abilityTracker.getAbility(spell.id).casts;
+  }
+
+  countEventsDuringBuff(buff: SpellInfo, eventType: string, cast?: SpellInfo | undefined) {
+    /**
+     * @param buff the spell object for the buff
+     * @param eventType the type of event that you want to search for. i.e. "cast", "begincast", EventType.Cast, EventType.BeginCast, etc.
+     * @param cast an optional cast spell object to count. Omit or leave undefined to count all casts
+     */
+    const events = this.getEventsDuringBuff(buff, eventType, cast);
+    return events.length;
+  }
+
+  getEventsDuringBuff(buff: SpellInfo, eventType: string, spell?: SpellInfo | undefined) {
+    /**
+     * @param buff the spell object for the buff
+     * @param eventType the type of event that you want to search for. i.e. "cast", "begincast", EventType.Cast, EventType.BeginCast, etc.
```

Comment:
This is implying it will work for any eventType when in fact it only works for Cast and BeginCast. If you want it to work for every event type obviously need to find something better than the switch statement. If you're fine with it being only those two types, you can actually restrict this in the type annotation like so: `eventType: 'cast' | 'begincast'`